src/hooks/tradingView/useTradingView.ts

```diff
@@ -75,7 +75,7 @@ export const useTradingView = ({
       tvWidgetRef.current = null;
       setIsChartReady(false);
     };
-  }, [getCandlesForDatafeed, isClientConnected, hasMarkets, selectedLocale, selectedNetwork]);
+  }, [getCandlesForDatafeed, isClientConnected, hasMarkets, selectedLocale, selectedNetwork, marketId]);
```

Comment:
I think we need an additional check as well. We don't want to re-initialize the chart when `marketId` changes, only on initial load.